packages/sdk-core/src/modules/database/BacktraceDatabase.ts

```diff
@@ -198,7 +198,11 @@ export class BacktraceDatabase implements BacktraceModule {
      */
     public async send() {
         for (let bucketIndex = 0; bucketIndex < this._databaseRecordContext.bucketCount; bucketIndex++) {
-            for (const record of this._databaseRecordContext.getBucket(bucketIndex)) {
+            // make a copy of records to not update the array after each remove
+            const records = [...this._databaseRecordContext.getBucket(bucketIndex)];
+
+            for (let recordIndex = 0; recordIndex < records.length; recordIndex++) {
+                const record = records[recordIndex];
```

Comment:
We can surely keep `foreach` if we are copying the array?

Reply:
Fair - My goal was to avoid foreach overall in the database operations, but we can keep the code simple. Changing to foreach